File: Lab2.3/Lab2.3/TextRow.cpp
#include "TextRow.h"
#include <cctype>

TextRow::TextRow() : text("default") {}

TextRow::TextRow(const std::string& str) : text(str) {}

std::string TextRow::getText() const {
    return text;
}

void TextRow::setText(const std::string& str) {
    text = str;
}

int TextRow::countDigits() const {
    int count = 0;
    for (char c : text) {
        if (std::isdigit(static_cast<unsigned char>(c)))
            count++;
    }
    return count;
}

File: Lab2.3/Lab2.3.1/TextRow.cs
namespace Lab5
{
    public class TextRow : INumberAnalysis
    {
        private string text;

        public TextRow()
        {
            text = "default";
        }

        public TextRow(string text)
        {
            this.text = text;
        }

        public string GetText() => text;
        public void SetText(string str) => text = str;

        public int CountDigits()
        {
            int count = 0;
            foreach (char c in text)
                if (char.IsDigit(c)) count++;
            return count;
        }
    }
}
